WIP: Refactor sqlitefuncts test to use test docker pattern; update dockerwrapper fixture

bonicim opened the pull request:
# Description

* Add warning to database unit tests that require sqlite3 to install sqlite3
* Check for docker image before pulling the image when using docker_wrapper fixture; DockerHub recently put pull-limits on free accounts. The dbutils unit tests repeatedly pull an image for each individual test, thus causing the tester to exceed the pull limit and preventing the tester to continue running the tests. See https://www.docker.com/increase-rate-limits


## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Ran the two related sqlite3 tests with and without sqlite3 installed to observe that the try-catch worked as expected.

# Checklist:

- [x] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

craig8 commented:
@bonicim another alternative is to do the same thing we do with the other dbs and download an sqlite3 container

https://hub.docker.com/r/nouchka/sqlite3/

Not saying one should, but it is a different approach that keeps everything running through the same code path

bonicim commented:
> @bonicim another alternative is to do the same thing we do with the other dbs and download an sqlite3 container
> 
> https://hub.docker.com/r/nouchka/sqlite3/
> 
> Not saying one should, but it is a different approach that keeps everything running through the same code path

Good point! I've been so used to treating sqlite3 as a second-class database that I didn't realize the same testing pattern can hold true as well. I can change the tests to match that pattern.

craig8 commented on services/core/SQLHistorian/tests/test_sqlitehistorian_unit.py:
```diff
@@ -1,8 +1,8 @@
 import subprocess
-
 import pytest
+
```
please reorganize these imports.  the order should be built-ins, add-ons, our stuff (generally)

easy to do with pycharm.  code menu I believe

craig8 commented:
@bonicim your tests did not run in my container.

bonicim commented:
Refactored test_sqlitefuncts.py to use volume container so that tests can be run on docker container environment in addition to running on a VM. 

Also fixed import order changes and addressed other format issues after running pylint. Results of running pylint locally shown below:

```
$ pylint services/core/SQLHistorian/tests/test_sqlitehistorian.py

+---------------------------+------------+
|message id                 |occurrences |
+===========================+============+
|protected-access           |4           |
+---------------------------+------------+
|missing-function-docstring |4           |
+---------------------------+------------+
|redefined-outer-name       |1           |
+---------------------------+------------+
|missing-module-docstring   |1           |
+---------------------------+------------+
|invalid-name               |1           |
+---------------------------+------------+

------------------------------------------------------------------
Your code has been rated at 7.11/10 (previous run: 7.11/10, +0.00)


$ pylint volttrontesting/platform/dbutils/test_sqlitefuncts.py

Messages
--------

+---------------------------+------------+
|message id                 |occurrences |
+===========================+============+
|redefined-outer-name       |41          |
+---------------------------+------------+
|missing-function-docstring |28          |
+---------------------------+------------+
|line-too-long              |15          |
+---------------------------+------------+
|unused-argument            |4           |
+---------------------------+------------+
|invalid-name               |3           |
+---------------------------+------------+
|missing-module-docstring   |1           |
+---------------------------+------------+


------------------------------------------------------------------
Your code has been rated at 5.18/10 (previous run: 4.87/10, +0.31)

```

bonicim commented:
Due to changes in #2622, which was recently merged in, this PR needs to be refactored to use the same test setup and patterns for testing SQL historians.

bonicim commented:
> Due to changes in #2622, which was recently merged in, this PR needs to be refactored to use the same test setup and patterns for testing SQL historians.

Closing this PR and will reopen in new PR to incorporate changes from #2622
